main.go

```diff
@@ -0,0 +1,312 @@
+)
+
+func init() {
+	runtime.Must(tenancyv1alpha1.AddToScheme(scheme.Scheme))
+	runtime.Must(corev1alpha1.AddToScheme(scheme.Scheme))
+	runtime.Must(apisv1alpha1.AddToScheme(scheme.Scheme))
+	runtime.Must(mongodbv1.AddToScheme(scheme.Scheme))
+}
+
+// label to place on synced objects to indicate which kcp logicalcluster they belong to
+const clusterLabel = "example-mongodb-multiclusterruntime/cluster"
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Fprintf(os.Stderr, "An error occured while running the example:\n%v\n", err)
+		os.Exit(1)
+	}
+}
+
+type reconciler struct {
+	log           *logr.Logger
+	targetClient  ctrlruntimeclient.Client
+	clusterGetter func(context.Context, string) (cluster.Cluster, error)
+}
+
+func run() error {
+	opts := struct {
+		// kubeconfig to connect to kcp provider workspace where APIExport exists
+		kcpkubeconfig string
+		// kubeconfig of target cluster where objects will be synced to
+		targetkubeconfig string
+	}{}
+
+	flag.StringVar(&opts.kcpkubeconfig, "kcp-kubeconfig", "", "kubeconfig to connect to kcp provider workspace where APIExport exists")
+	flag.StringVar(&opts.targetkubeconfig, "target-kubeconfig", "", "kubeconfig of target cluster where objects will be synced to")
+
+	zapOpts := kubezap.Options{
+		Development: true,
+	}
+	zapOpts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	if opts.kcpkubeconfig == "" || opts.targetkubeconfig == "" {
+		return errors.New("both --kcp-kubeconfig and --target-kubeconfig must be specified")
+	}
+
+	logger := kubezap.New(kubezap.UseFlagOptions(&zapOpts))
+	ctrl.SetLogger(logger)
+	log := logger.WithName("setup")
+
+	// setup downstream reconciliation
+	kcpConfig, err := clientcmd.BuildConfigFromFlags("", opts.kcpkubeconfig)
+	if err != nil {
+		return err
+	}
+
+	provider, err := apiexport.New(kcpConfig, apiexport.Options{})
+	if err != nil {
+		return err
+	}
+
+	mgr, err := mcmanager.New(kcpConfig, provider, manager.Options{
+		Logger: logger.WithName("manager"),
+	})
+	if err != nil {
+		return err
+	}
+
+	// setup upstream reconciliation
+	targetConfig, err := clientcmd.BuildConfigFromFlags("", opts.targetkubeconfig)
+	if err != nil {
+		return err
+	}
+	targetClient, err := ctrlruntimeclient.New(targetConfig, ctrlruntimeclient.Options{})
+	if err != nil {
+		return err
+	}
+
+	// build the reconciler
+	rec := &reconciler{
+		log:           &logger,
+		targetClient:  targetClient,
+		clusterGetter: mgr.GetCluster,
+	}
+	if err := rec.SetupWithManager(mgr, targetConfig, log); err != nil {
+		return err
+	}
+	log.Info("Setup complete")
+
+	log.Info("Starting provider and manager")
+	g, ctx := errgroup.WithContext(context.Background())
+	g.Go(func() error { return wrapError("provider", provider.Run(ctx, mgr)) })
+	g.Go(func() error { return wrapError("manager", mgr.Start(ctx)) })
+
+	return g.Wait()
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req mcreconcile.Request) (reconcile.Result, error) {
+	clustername := req.ClusterName
```

Comment:
```suggestion
	clusterName := req.ClusterName
```

For consistency with other projects.